scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -231,18 +284,79 @@
     (async/chan 1))
 
   (pending-offers [this pool-name]
-    (let [nodes @current-nodes-atom
-          pods (all-pods this @all-pods-atom)
-          offers-all-pools (generate-offers this nodes pods)
-          ; TODO: We are generating offers for every pool here, and filtering out only offers for this one pool.
-          ; TODO: We should be smarter here and generate once, then reuse for each pool, instead of generating for each pool each time and only keeping one
-          offers-this-pool (get offers-all-pools pool-name)]
-      (log/info "Generated" (count offers-this-pool) "offers for pool" pool-name "in compute cluster" name
-                     (into [] (map #(into {} (select-keys % [:hostname :resources])) offers-this-pool)))
-      offers-this-pool))
+    (let [pods (add-starting-pods this @all-pods-atom)
+          num-waiting-synthetic-pods (num-waiting-synthetic-pods this pods pool-name)
+          nodes @current-nodes-atom]
+      (log/info "In" name "compute cluster, got asked for pending offers for pool" pool-name
+                {:total-pods (count pods)
+                 :synthetic-pods num-waiting-synthetic-pods
+                 :nodes (count nodes)})
+      (if (pos? num-waiting-synthetic-pods)
+        (do
```

Comment:
We should limit how long we avoid generating offers rather than being open to the possibility of waiting indefinitely because synthetic pods didn't get scheduled.

Reply:
Good call, will do

Reply:
Changed it so that only "recent" synthetic pods count, the idea being that if a synthetic pod was created > X seconds ago (e.g. X might be 180), then we stop considering it here as a "waiting" synthetic pod, which means it will no longer hold up offer generation.